src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -89,7 +89,8 @@ CPP_template(typename Range1, typename Range2, typename LessThan,
              typename CompatibleRowAction, typename FindSmallerUndefRangesLeft,
              typename FindSmallerUndefRangesRight,
              typename ElFromFirstNotFoundAction = decltype(noop),
-             typename CheckCancellation = decltype(noop))(
+             typename CheckCancellation = decltype(noop),
+             typename CoverUndefRanges = std::true_type)(
```

Comment:
Add some documentation to this very particular argument.

Reply:
In particular add that this only works if there is UNDEF in a left join column that is not the last one.